### PR TITLE
Friendly Fire Four: A New NT

### PR DESCRIPTION
**Pull request by bathosbathtime**
This Pull Request adds the Nanotrasen RPG, the low yield rocket, the foam rocket, and the EMP rocket.
Three RPGs have been added to the maps currently in rotation.  Also Metaclub and Bagel, I don't recall if Metaclub is in rotation but I added RPGs to it.  Two of the three rocket launchers have been placed in secure armory with the and one is in the security office.  Packed station has one in the armory.  These Nanotrasen branded rocket launchers spawn with the foam rocket equiped.  All three rocket types have been added to the ammolathe.

* Low Yield: Low penetration medium damage rocket.  Made so security can handle target rich environments without breaching.
* Foam: Extremely low damage rocket that stuns a single target.  Designed so I can have the rockets spawn in without giving someone a room killing weapon.  Can also be used as worse stun shells.
* EMP: Low damage rocket that creates an emp at detonation location.

compiled and tested locally

:cl:
 * experiment: Adds the Nanotrasen RPG to the brig.
 * rscadd: Adds the Low Yield Rocket, the Foam rocket, and the EMP rocket.

### CODE REVIEW

File: code/modules/projectiles/guns/projectile/rocketlauncher.dm [hunk 1]
```diff
@@ -53,6 +53,64 @@
 			return(SUICIDE_ACT_BRUTELOSS)
 	return
 
+/obj/item/weapon/gun/projectile/rocketlauncher/nanotrason
```
Comment (boy2mantwicethefam) [inline code/modules/projectiles/guns/projectile/rocketlauncher.dm]: nanotrasen

Comment (bathosbathtime) [reply code/modules/projectiles/guns/projectile/rocketlauncher.dm]: whoops

**Comment by SonixApache** [comment]
Why a fugly blue RPG-7 instead of that lovely LAW sprite people posted?

**Comment by DrSnips** [comment]
Why not put the AMMO in the AMMOlathe?
Edit: I am retard and can't read.

**Comment by bathosbathtime** [comment]
It is a reskinned RPG as it behaves like the syndicate RPG.
They can even use the ammunition from the other if you happen to get your hands on any, which is why I used the same sprite recolored.

File: code/modules/projectiles/guns/projectile/rocketlauncher.dm [hunk 2]
```diff
@@ -53,6 +53,64 @@
...
+		user.visible_message("<span class='danger'>[user] jams down \the [src]'s trigger before noticing it isn't loaded and starts bashing \his head in with it! It looks like \he's trying to commit suicide.</span>")
+		return(SUICIDE_ACT_BRUTELOSS)
+	else //Needed to get that shitty default suicide_act out of the way
+		user.visible_message("<span class='danger'>[user] fiddles with \the [src]'s safeties and suddenly aims it at \his feet! It looks like \he's trying to commit suicide.</span>")
+		spawn(10) //RUN YOU IDIOT, RUN
+			explosion(src.loc, -1, 1, 4, 8)
+			if(src) //Is the rocket launcher somehow still here ?
+				qdel(src) //This never happened
+			return(SUICIDE_ACT_BRUTELOSS)
+	return
```
Comment (SonixApache) [inline code/modules/projectiles/guns/projectile/rocketlauncher.dm]: If this and the `attack()` are the same as the normal RPG, you don't have to define them

File: code/modules/projectiles/projectile/rocket.dm [hunk 3]
```diff
@@ -26,7 +27,111 @@
...
+			picked_up_speed--
+		if(dist_x > dist_y)
+			bresenham_step(dist_x,dist_y,dx,dy)
+		else
+			bresenham_step(dist_y,dist_x,dy,dx)
+		if(linear_movement)
+			update_pixel()
+			pixel_x = PixelX
+			pixel_y = PixelY
+		sleep(picked_up_speed)
```
Comment (SonixApache) [inline code/modules/projectiles/projectile/rocket.dm:94]: You don't need to copypaste the `process_step()` if it does the same as the parent class.

File: code/modules/projectiles/projectile/rocket.dm [hunk 4]
```diff
@@ -26,7 +27,111 @@
...
+	name = "low yield rocket"
+	icon_state = "rpground"
+	explosive = 0
+	damage = 30
+	stun = 5
+	weaken = 5
+	damage_type = BRUTE
+	nodamage = 0
+	flag = "bullet"
+	fire_sound = 'sound/weapons/rocket.ogg'
```
Comment (SonixApache) [inline code/modules/projectiles/projectile/rocket.dm]: defining `fire_sound` when it's the same as the parent class is also unnecessary.

**Comment by SonixApache** [comment]
Sad that the foam rocket is essentialy a nerf rocket, it'd be cool to foam people and mechs to the ground with it.
Light explosions still have a chance to breach, if I'm not mistaken.

Remove your mapping changes (which undo some recent changes) and put your RPGs and whatever ammo you want to add in the vault lockers.

https://github.com/vgstation-coders/vgstation13/blob/82222271d15a05ce2820f3815a5b1518dbe726b6/code/game/objects/structures/crates_lockers/closets/secure/vault.dm#L33-L39

File: maps/metaclub.dmm [hunk 5]
```diff
@@ -9242,7 +9242,7 @@
 "dvL" = (/obj/structure/grille,/obj/structure/window/reinforced,/obj/structure/window/reinforced{dir = 4},/obj/structure/window/reinforced{dir = 1},/obj/structure/window/full/reinforced,/turf/simulated/shuttle/plating/vox,/area/shuttle/vox/station)
 "dvM" = (/obj/machinery/computer/security,/turf/simulated/shuttle/floor/vox{icon_state = "floor4"},/area/shuttle/vox/station)
 "dvN" = (/obj/machinery/computer/shuttle_control/vox,/turf/simulated/shuttle/floor/vox{icon_state = "floor4"},/area/shuttle/vox/station)
-"dvO" = (/obj/machinery/computer/crew{holomap_filter = HOLOMAP_FILTER_VOX},/turf/simulated/shuttle/floor/vox{icon_state = "floor4"},/area/shuttle/vox/station)
+"dvO" = (/obj/machinery/computer/crew{holomap_filter = 16},/turf/simulated/shuttle/floor/vox{icon_state = "floor4"},/area/shuttle/vox/station)
```
Comment (SonixApache) [inline maps/metaclub.dmm:9245]: 🤔

**Comment by SGuivenne** [comment]
Any chance of adding some kind of chem rockets in?  Rockets that spew a chemical payload.  Would mostly be paper intended to make metal foam rockets but could theoretically mean you can DIY shitty napalm or HE rockets if you really worked at it or that you could make LSD rockets.

**Comment by iFlashYou** [comment]
>security buffs

**Comment by FudgePucker** [comment]
Maybe make the foam rocket, instead of being a "toy" rocket, have it be a "foam" rocket, as in cleaning grenade foam. It could be used to non lethally calm down a room by making people slip on the ground.

You could also use a screwdriver on the rocket to remove the pay load and add in a loaded chemical grenade casing for the idea that @SGuivenne has.

**Comment by FudgePucker** [comment]
Also make the nukeop RPG start with the foam rocket if you do not go with my idea.
I swear to god people that pick that thing up as Ops will be the downfall of your operation 110% of the time. I have nuke op set to never just because of them.

**Comment by bathosbathtime** [comment]
I will be re-doing this completely with much better code and far less map changes.

**Pull request closed without merging**